docs/tutorials/integrations/nginx.md

```diff
@@ -0,0 +1,135 @@
+is called a **man in the middle attack**.
+
+Even if I'm not able to impersonate the Conjur server, I could still learn your
+secrets by joining your network and listening for traffic coming and going from
+the Conjur server. This is called **passive surveillance**.
+
+### TLS defeats passive and active attacks
+
+Transport Layer Security allows your client to verify that it's talking to the
+real Conjur server, and it uses standard secure technology to encrypt your
+secrets in transit. This means:
+
+* Your Conjur server will be `https:` instead of `http:`, just like a secure
+  website
+* Because the client knows it's talking to the real server, the "man in the
+  middle" will be exposed and the client won't leak any secret information
+* Since the traffic to and from Conjur is scrambled using secure encryption,
+  passive listeners on your network can't learn anything about the contents of
+  your secrets
+
+For these reasons, it is crucial to set up TLS correctly when you deploy Conjur.
+
+### Protect Your Secrets
+
+Do not leave this to chance: even a small flaw can totally compromise your
+secrets. Setting up Conjur and TLS without the appropriate expertise is like
+packing your own parachute and jumping out of a plane.
+
+That doesn't mean you should close the tab and walk away. It means you should get
+in touch with us and your own security team so we can ensure that you can employ
+Conjur successfully.
+
+[authn]: https://www.conjur.org/api.html#authentication-authenticate-post
+[secret-get]: https://www.conjur.org/api.html#secrets-retrieve-a-secret-get
+
+## Using NGINX To Proxy Traffic With TLS To Conjur
+
+[NGINX][NGINX] is a web proxy that's Free Software and easy to configure. This
+tutorial will show you how to use Docker to install Conjur and NGINX and
+configure them to provide Conjur with TLS.
+
+### Prerequisites
+
+This tutorial requires Docker and a terminal application. Prepare by following
+the prerequisite instructions found on [Install Conjur][prerequisites].
+
+Additionally, you will need the tutorial files from the Conjur source code
+repository. Here's how you get them:
+
+1. Install Git: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
+1. Clone the Conjur repository
+   
+   In your terminal application, run:
+   ```sh-session
+   $ git clone git@github.com:cyberark/conjur.git
+   ```
+   This will create a folder called `conjur` in the current directory.
+
+### The Good Part
+
+The tutorial script will install Conjur and NGINX, configure them to work
+together, and connect a Conjur client to the running server. This is a full
+end-to-end working installation to allow you to see how the pieces connect
+together.
+
+```sh-session
+$ # start the Conjur+NGINX tutorial servers
+$ cd conjur/docs/tutorials/integrations/nginx
+$ ./start.sh
+```
+
+Take a look at the logs for the Conjur or NGINX servers:
+
+```sh-session
+$ docker-compose logs conjur
+$ docker-compose logs proxy
```

Comment:
It could be clearer that the user should be doing this from the same `nginx` dir as above

Reply:
Also, something is weird here. After running through the tutorial and coming back to this command, here is what I see:
```
$ docker-compose logs conjur
Attaching to nginx_conjur_1
conjur_1    | => Booting Puma
conjur_1    | => Rails 4.2.8 application starting in production on http://0.0.0.0:80
conjur_1    | => Run `rails server -h` for more startup options
conjur_1    | => Ctrl-C to shutdown server
conjur_1    | [1] Puma starting in cluster mode...
conjur_1    | [1] * Version 3.8.2 (ruby 2.2.8-p477), codename: Sassy Salamander
conjur_1    | [1] * Min threads: 5, max threads: 5
conjur_1    | [1] * Environment: production
conjur_1    | [1] * Process workers: 2
conjur_1    | [1] * Preloading application
conjur_1    | [1] * Listening on tcp://0.0.0.0:80
```

Reply:
Note in particular that it doesn't seem to be working through port 443 or over https - can you be clearer about what they should be looking for in the logs to verify TLS? I'm not seeing it in the logs

Reply:
I did get a 200 response (and the "Your server is running!" page) when I ran `curl -i https://localhost:443 --cacert tls/nginx.crt`, however - and a "failure to connect" message when trying localhost:80.

Reply:
So what's happening here is the Conjur server is listening on port 80, but it isn't exposing that port except on the local Docker network. Meanwhile, the nginx container is exposing its port 443 and proxying through to Conjur.

I will add this explanation to the tutorial!